src/main/java/org/assertj/core/api/Descriptable.java

```diff
@@ -23,7 +24,7 @@
  * @author Yvonne Wang
  * @author Mikhail Mazursky
  */
-public interface Descriptable<SELF> {
+public interface Descriptable<SELF extends Descriptable<SELF>> {
```

Comment:
Just FYI this change is not a backwards compatible change. Why is it needed to force that `SELF` extends from `Descriptable`?

Reply:
This is true. However, it's not really a "SELF" type if it doesn't extend itself? I think it was probably an oversight when the type was originally created - I can't think of when you'd have a use case that SELF is not an instance of `Descriptable`.

On the other hand, it is not essential to the main purpose of this PR, so I can leave this change out of this PR for further discussion. The rest of the PR should (hopefully!) be less controversial.

Reply:
Let's take the safest approach and not extend from Descriptable.

Reply:
Latest push has removed the constraint on `SELF`.